sdk/nodejs/apps/v1/statefulSet.ts

```diff
@@ -25,133 +25,131 @@ import * as utilities from "../../utilities";
- *                     ports: [{
- *                         containerPort: 80,
- *                         name: "web",
- *                     }],
- *                     volumeMounts: [{
- *                         name: "www",
- *                         mountPath: "/usr/share/nginx/html",
- *                     }],
+ *         spec: {
+ *             containers: [{
+ *                 image: "k8s.gcr.io/nginx-slim:0.8",
+ *                 name: "nginx",
+ *                 ports: [{
+ *                     containerPort: 80,
+ *                     name: "web",
  *                 }],
- *             },
+ *                 volumeMounts: [{
+ *                     mountPath: "/usr/share/nginx/html",
+ *                     name: "www",
+ *                 }],
+ *             }],
+ *             terminationGracePeriodSeconds: 10,
  *         },
- *         volumeClaimTemplates: [{
- *             metadata: {
- *                 name: "www",
- *             },
- *             spec: {
- *                 accessModes: ["ReadWriteOnce"],
- *                 storageClassName: "my-storage-class",
- *                 resources: {
- *                     requests: {
- *                         storage: "1Gi",
- *                     },
+ *     },
+ *     volumeClaimTemplates: [{
+ *         metadata: {
+ *             name: "www",
+ *         },
+ *         spec: {
+ *             accessModes: ["ReadWriteOnce"],
+ *             resources: {
+ *                 requests: {
+ *                     storage: "1Gi",
  *                 },
  *             },
- *         }],
- *     },
- * });
+ *             storageClassName: "my-storage-class",
+ *         },
+ *     }],
+ * }});
  * ```
  * ### Create a StatefulSet with a user-specified name
  *
  * ```typescript
  * import * as pulumi from "@pulumi/pulumi";
  * import * as kubernetes from "@pulumi/kubernetes";
  *
- * const nginxService = new kubernetes.core.v1.Service("nginxService", {
+ * const service = new kubernetes.core.v1.Service("service", {
  *     metadata: {
- *         name: "nginx",
  *         labels: {
  *             app: "nginx",
  *         },
+ *         name: "nginx",
  *     },
  *     spec: {
+ *         clusterIP: "None",
  *         ports: [{
- *             port: 80,
  *             name: "web",
+ *             port: 80,
  *         }],
- *         clusterIP: "None",
  *         selector: {
  *             app: "nginx",
  *         },
  *     },
  * });
- * const wwwStatefulSet = new kubernetes.apps.v1.StatefulSet("wwwStatefulSet", {
+ * const statefulset = new kubernetes.apps.v1.StatefulSet("statefulset", {
  *     metadata: {
  *         name: "web",
  *     },
  *     spec: {
+ *         replicas: 3,
  *         selector: {
  *             matchLabels: {
  *                 app: "nginx",
  *             },
  *         },
- *         serviceName: nginxService.metadata.name,
- *         replicas: 3,
+ *         serviceName: service.metadata.apply(metadata => metadata?.name),
```

Comment:
Side note: I'm guessing the `.metadata.name` property isn't marked as required in the schema, which results in the ugly apply rather than the lifted form.

Reply:
Yeah interestingly this is one of the reasons to use pulumi convert over kube2pulumi here - since it allows us to embed resource references and produce more pulumi-esque code.